test/gromov/test_lowrank.py

```diff
@@ -0,0 +1,120 @@
+
+    a = ot.unif(n_samples)
+    b = ot.unif(n_samples)
+
+    Q, R, g, log = ot.gromov.lowrank_gromov_wasserstein(X_s, X_t, a, b, reg=0.1, log=True, rescale_cost=False)
+    P = log["lazy_plan"][:]
+
+    # check constraints for P
+    np.testing.assert_allclose(a, P.sum(1), atol=1e-04)
+    np.testing.assert_allclose(b, P.sum(0), atol=1e-04)
+
+    # check if lazy_plan is equal to the fully computed plan
+    P_true = np.dot(Q, np.dot(np.diag(1 / g), R.T))
+    np.testing.assert_allclose(P, P_true, atol=1e-05)
+
+    # check warn parameter when low rank GW algorithm doesn't converge
+    with pytest.warns(UserWarning):
+        ot.gromov.lowrank_gromov_wasserstein(
+            X_s, X_t, a, b, reg=0.1, stopThr=0, numItermax=1, warn=True, warn_dykstra=False
+        )
+
+    # check warn parameter when Dykstra algorithm doesn't converge
+    with pytest.warns(UserWarning):
+        ot.gromov.lowrank_gromov_wasserstein(
+            X_s, X_t, a, b, reg=0.1, stopThr_dykstra=0, numItermax_dykstra=1, warn=False, warn_dykstra=True
+        )
+
+
+@pytest.mark.parametrize(("alpha, rank"), ((0.8, 2), (0.5, 3), (0.2, 6), (0.1, -1)))
+def test_lowrank_gromov_wasserstein_alpha_error(alpha, rank):
+    # Test warning for value of alpha and rank
+    n_samples = 20  # nb samples
+    mu_s = np.array([0, 0])
+    cov_s = np.array([[1, 0], [0, 1]])
+
+    X_s = ot.datasets.make_2D_samples_gauss(n_samples, mu_s, cov_s, random_state=1)
+    X_t = X_s[::-1].copy()
+
+    a = ot.unif(n_samples)
+    b = ot.unif(n_samples)
+
+    with pytest.raises(ValueError):
+        ot.gromov.lowrank_gromov_wasserstein(X_s, X_t, a, b, reg=0.1, rank=rank, alpha=alpha, warn=False)
+
+
+@pytest.mark.parametrize(("gamma_init"), ("rescale", "theory"))
+def test_lowrank_wasserstein_gamma_init(gamma_init):
+    # Test lr sinkhorn with different init strategies
+    n_samples = 20  # nb samples
+    mu_s = np.array([0, 0])
+    cov_s = np.array([[1, 0], [0, 1]])
+
+    X_s = ot.datasets.make_2D_samples_gauss(n_samples, mu_s, cov_s, random_state=1)
+    X_t = X_s[::-1].copy()
+
+    a = ot.unif(n_samples)
+    b = ot.unif(n_samples)
+
+    Q, R, g, log = ot.gromov.lowrank_gromov_wasserstein(X_s, X_t, a, b, reg=0.1, gamma_init=gamma_init, log=True)
+    P = log["lazy_plan"][:]
+
+    # check constraints for P
+    np.testing.assert_allclose(a, P.sum(1), atol=1e-04)
+    np.testing.assert_allclose(b, P.sum(0), atol=1e-04)
+
+
+@pytest.skip_backend('tf')
+def test_lowrank_gromov_wasserstein_backends(nx):
+    # Test low rank sinkhorn for different backends
+    n_samples = 20  # nb samples
+    mu_s = np.array([0, 0])
+    cov_s = np.array([[1, 0], [0, 1]])
+
+    X_s = ot.datasets.make_2D_samples_gauss(n_samples, mu_s, cov_s, random_state=1)
+    X_t = X_s[::-1].copy()
+
+    a = ot.unif(n_samples)
+    b = ot.unif(n_samples)
+
+    ab, bb, X_sb, X_tb = nx.from_numpy(a, b, X_s, X_t)
+
+    Q, R, g, log = ot.gromov.lowrank_gromov_wasserstein(X_sb, X_tb, ab, bb, reg=0.1, log=True)
```

Comment:
need to test that results are consistent across backend taking e.g numpy as reference backend, e.g modifying in place `test_lowrank_gromov_wasserstein` .